app/controllers/events_controller.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+class EventsController < ApplicationController
+  before_action :signed_in?, :set_event, only: %i[show edit update]
+
+  # GET /events
+  def index
+    if current_user.present?
+      @upcoming_events = Event.upcoming_events
+      @prev_events = Event.prev_events
+    else
+      flash.now[:alert] = 'You are not Signed in'
+      redirect_to signin_path
+    end
```

Comment:
Do it in `before_action` method

Reply:
https://github.com/Israel-Laguan/Private-Events/blob/6052e77d3197ccd68c6fb59832431ea96e433f7f/app/controllers/events_controller.rb#L7-L10
Done!